themes/bootstrap/static/js/filter.js

```diff
@@ -0,0 +1,228 @@
+ *  description, license, name, languages, and owners as strings.
+ */
+function populateToolsKeywords(tools) {
+  for (let i = 0; i < tools.length; i++) {
+    let keywords = new Set();
+    const pattern = /[\s\-_.,!?]/;
+    const descriptionArr = tools[i].description ? tools[i].description.split(pattern) : [];
+    const licenseArr = tools[i].license ? tools[i].license.split(pattern) : [];
+    const nameArr = tools[i].name ? tools[i].name.split(pattern) : [];
+
+    descriptionArr.map(w => keywords.add(w.toLowerCase()));
+    licenseArr.map(w => keywords.add(w.toLowerCase()));
+    nameArr.map(w => keywords.add(w.toLowerCase()));
+
+    // Note that owner and language are already arrays -- no processing needed.
+    tools[i].owner.map(w => keywords.add(w.toLowerCase()));
+    tools[i].language.map(w => keywords.add(w.toLowerCase()));
+
+    tools[i]['keywords'] = keywords;
+  }
+}
+
+
+/** Given a the toolsData and set of languages and owners, populate the tools table
+ *  on the DOM. If languages is empty, use all languages. If owners is empty,
+ *  use all owners.
+ */
+function fillToolsTable(tools, selectedLanguages, selectedOwners) {
+    $('#tools-table').empty();
+    let numMatchingTools = 0;
+    for (let tool of tools) {
+        const languageMatch = tool.language.some(lang => selectedLanguages.has(lang));
+        const ownerMatch = tool.owner.some(o => selectedOwners.has(o));
+        if (
+          (selectedLanguages.size === 0 || languageMatch)
+          && (selectedOwners.size === 0 || ownerMatch)
+          && (!excludeDormant || !tool.dormant)
+          && (!excludeArchived || !tool.archived)
+          && (searchMatch(searchTerm, tool.keywords))
+        ) {
+            numMatchingTools++;
+            $('#tools-table').append(makeTableRow(tool));
+        }
+    }
+    updateSearchResultMessage(numMatchingTools)
+}
+
+
+/** If searchTerm is in keywords (even as a partial match), return true. */
+function searchMatch(searchTerm, keywords) {
+  for (let word of keywords) {
+    if (word.toLowerCase().includes(searchTerm)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+
+/** Given a tool, return the HTML string to represent it in the tools table. */
+function makeTableRow(tool) {
+    return (
+    `<tr>
+      <td>
+        <p>
+          <strong>
+            <a title="Project" href="${tool.url}">${tool.name}</a>
+          </strong>&ensp;
+          <span style="display: ${tool.dormant ? 'visible' : 'none'};" class="tooltip">
+            <em class="bi bi-moon-stars"></em>
+            <span class="tooltip-text">project is dormant</span>
+          </span>
+          <span style="display: ${tool.archived ? 'visible' : 'none'}" class="tooltip">
+            <em class="bi bi-archive"></em>
+            <span class="tooltip-text">project has been archived</span>
+          </span><br>
+          <span style="display: ${tool.owner.length > 0 ? 'visible' : 'none'}>
```

Comment:
Add missing double colon:
```suggestion
          <span style="display: ${tool.owner.length > 0 ? 'visible' : 'none'}">
```